tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+        sipForward = makeRandomSipForwardTransform(4)
+        converted = ScaledPolynomialTransform.convert(sipForward)
+        self.checkTransformEquivalence(sipForward, converted)
+
+    def testConvertSipReverse(self):
+        """Test that we can convert a SipReverseTransform to a ScaledPolynomialTransform.
+        """
+        sipReverse = makeRandomSipReverseTransform(4)
+        converted = ScaledPolynomialTransform.convert(sipReverse)
+        self.checkTransformEquivalence(sipReverse, converted)
+
+
+class SipForwardTransformTestCase(lsst.utils.tests.TestCase, TransformTestMixin):
+
+    def setUp(self):
+        numpy.random.seed(50)
+
+    def makeRandom(self):
+        return makeRandomSipForwardTransform(4)
+
+    def testConstruction(self):
+        poly = makeRandomPolynomialTransform(4, sip=True)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = SipForwardTransform(crpix, cd, poly)
+        self.checkTransformEquivalence(
+            sip,
+            lambda p: cd((p - crpix) + poly(lsst.afw.geom.Point2D(p- crpix)))
+        )
+
+    def testConvertPolynomial(self):
+        poly = makeRandomPolynomialTransform(4)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = lsst.meas.astrom.SipForwardTransform.convert(poly, crpix, cd)
+        self.checkTransformEquivalence(sip, poly)
+
+    def testConvertScaledPolynomialManual(self):
+        scaled = makeRandomScaledPolynomialTransform(4)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = lsst.meas.astrom.SipForwardTransform.convert(scaled, crpix, cd)
+        self.checkTransformEquivalence(sip, scaled)
+
+    def testConvertScaledPolynomialAutomatic(self):
+        scaled = makeRandomScaledPolynomialTransform(4)
+        sip = lsst.meas.astrom.SipForwardTransform.convert(scaled)
+        self.checkTransformEquivalence(sip, scaled)
+
+    def testMakeWcs(self):
+        fwd = self.makeRandom()
+        rev = SipReverseTransform(  # this isn't actually the inverse of fwd, but that doesn't matter
+            fwd.getPixelOrigin(),
+            fwd.getCDMatrix(),
+            makeRandomPolynomialTransform(4)
+        )
+        crval = lsst.afw.geom.Point2D(45.0, 45.0)
+        wcs = lsst.meas.astrom.makeWcs(
+            fwd, rev,
+            lsst.afw.coord.IcrsCoord(crval, lsst.afw.geom.degrees)
+        )
+        def t1(p):
+            sky = lsst.afw.coord.IcrsCoord(crval + lsst.afw.geom.Extent2D(p), lsst.afw.geom.degrees)
+            return rev(wcs.skyToIntermediateWorldCoord(sky))
+        def t2(p):
+            sky = lsst.afw.coord.IcrsCoord(crval + lsst.afw.geom.Extent2D(p), lsst.afw.geom.degrees)
+            return wcs.skyToPixel(sky)
+        self.checkTransformEquivalence(t1, t2)
```

Comment:
That's actually a lot harder - it requires actually inverting the polynomial part of `rev` (note that `fwd` isn't actually the inverse of `rev` right now).  I do essentially that down in `testFromGrid`.